aws/auth/mutable_static_creds_provider.cc

```diff
@@ -15,137 +15,36 @@
 #include <limits>
 
 namespace {
-#ifdef DEBUG
-  thread_local aws::auth::DebugStats debug_stats;
-
-  void update_step(std::uint64_t& step, std::uint64_t& outcome) {
-    step++;
-    outcome++;
-    debug_stats.attempts_++;
-  }
-
-  void update_step(std::uint64_t& step) {
-    std::uint64_t ignored = 0;
-    update_step(step, ignored);
-  }
-#else
-#define update_step(...)
-#endif
+  thread_local aws::auth::VersionedCredentials current_creds;
```

Comment:
I think a short comment about the role of the thread local current creds and how its version is compared against the global version would make it easier to follow.